File: 백준/Silver/3273. 두 수의 합/두 수의 합.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.*;

public class Main {

    private static int n, x;
    private static int[] nums;

    public static void main(String[] args) throws IOException {
        init();
        findAnswer();
    }

    private static void init() throws IOException {
        BufferedReader input = new BufferedReader(new InputStreamReader(System.in));
        n = Integer.parseInt(input.readLine());
        nums = new int[n];
        StringTokenizer tokenizer = new StringTokenizer(input.readLine());
        for (int i = 0; i < n; i++) {
            nums[i] = Integer.parseInt(tokenizer.nextToken());
        }
        Arrays.sort(nums);
        x = Integer.parseInt(input.readLine());
    }

    private static void findAnswer() {
        int left = 0;
        int right = nums.length - 1;
        int count = 0;
        while (left < right) {
            int sum = nums[left] + nums[right];
            if (sum == x) {
                if (nums[left] == nums[left + 1]) {
                    left ++;
                } else {
                    right--;
                }
                count++;
            }
            if (sum < x) {
                left++;
            }
            if (sum > x) {
               right--;
            }
        }
        System.out.println(count);
    }
}

File: 백준/Silver/3273. 두 수의 합/두 수의 합.cc
#include <bits/stdc++.h>
using namespace std;
int n, x;
int a[10000001];
int cnt;
int main() {
    cin >> n;
    for (int i = 0; i < n; i++) {
        cin >> a[i];
    }
    cin >> x;
    sort(a, a + n);
    int l = 0;
    int r = n - 1;
    while (l < r) {
        int sum = a[l] + a[r];
        if (sum == x) {
            cnt++;
            l++;
            r--;
        } else if (sum < x) {
            l++;
        } else {
            r--;
        }
    }

    cout << cnt << '\n';
    return 0;
}